pkg/e2e/verify/storage.go

```diff
@@ -18,17 +26,284 @@ func init() {
+		cfg.ClaimSize = "2Gi"
+	}
+
+	if len(cfg.NamePrefix) == 0 {
+		cfg.NamePrefix = "pvc-"
+	}
+
+	if cfg.VolumeMode != nil && *cfg.VolumeMode == "" {
+		log.Printf("Warning: Making PVC: VolumeMode specified as invalid empty string, treating as nil")
+		cfg.VolumeMode = nil
+	}
+
+	return &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:         cfg.Name,
+			GenerateName: cfg.NamePrefix,
+			Namespace:    ns,
+			Annotations:  cfg.Annotations,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			Selector:    cfg.Selector,
+			AccessModes: cfg.AccessModes,
+			Resources: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse(cfg.ClaimSize),
+				},
+			},
+			StorageClassName: cfg.StorageClassName,
+			VolumeMode:       cfg.VolumeMode,
+		},
+	}
+}
+
+// createPVC creates the PVC resource. Fails test on error.
+func createPVC(h *helper.H, ns string, pvc *corev1.PersistentVolumeClaim) (*corev1.PersistentVolumeClaim, error) {
+	pvc, err := h.Kube().CoreV1().PersistentVolumeClaims(ns).Create(context.TODO(), pvc, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("PVC Create API error: %v", err)
+	}
+	return pvc, nil
+}
+
+// This function Deletes the PVC.
+func deletePersistentVolumeClaim(h *helper.H, pvcName string, ns string) error {
+	if h != nil && len(pvcName) > 0 {
+		log.Printf("Deleting PersistentVolumeClaim %q", pvcName)
+		err := h.Kube().CoreV1().PersistentVolumeClaims(ns).Delete(context.TODO(), pvcName, metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("PVC Delete API error: %v", err)
+		}
+	}
+	return nil
+}
+
+// Config is a struct containing all arguments for creating a pod.
+// SELinux testing requires to pass HostIPC and HostPID as boolean arguments.
+type Config struct {
+	NS                     string
+	PVCs                   []*corev1.PersistentVolumeClaim
+	PVCsReadOnly           bool
+	InlineVolumeSources    []*corev1.VolumeSource
+	IsPrivileged           bool
+	Command                string
+	HostIPC                bool
+	HostPID                bool
+	SeLinuxLabel           *corev1.SELinuxOptions
+	FsGroup                *int64
+	ImageID                int
+	PodFSGroupChangePolicy *corev1.PodFSGroupChangePolicy
+}
+
+// makeTestPod returns a pod definition based on the namespace. The pod references the PVC's
+// name.
+func makeTestPod(ns string, pvclaims []*corev1.PersistentVolumeClaim, isPrivileged bool) *corev1.Pod {
+	podSpec := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "pvc-tester-",
+			Namespace:    ns,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "dummy",
+					Image:   "registry.access.redhat.com/ubi8/ubi-minimal",
+					Command: []string{"/bin/sh"},
+					Args:    []string{"-c", "touch /mnt/volume1/hello.txt; sleep 3600"},
```

Comment:
can we echo a string to the file and read the string back? this would be a simple write/read test of the file system...

Reply:
I added that change, thanks for the suggestion @dustman9000!